presto-main/src/main/java/com/facebook/presto/metadata/TableLayoutResult.java

```diff
@@ -44,11 +47,14 @@ public TableLayout getLayout()
 
     public boolean hasAllOutputs(TableScanNode node)
     {
-        List<ColumnHandle> nodeColumnHandles = node.getOutputSymbols().stream()
+        if (!layout.getColumns().isPresent()) {
```

Comment:
this might be potentially a different table scan (e.g: for other table). Maybe there is a way to ensure `layout handle` is for particular `table handle` (I haven't seen any trivial way to validate that).